File: DSA/NarindVerma_500121898/Day6Q1.py
def findDuplicate(nums):
    # Phase 1: Detect intersection point in the cycle
    slow = fast = nums[0]
    while True:
        slow = nums[slow]
        fast = nums[nums[fast]]
        if slow == fast:
            break

    # Phase 2: Find entrance to cycle
    slow = nums[0]
    while slow != fast:
        slow = nums[slow]
        fast = nums[fast]

    return slow

# Examples
print(findDuplicate([1, 3, 4, 2, 2]))  # Output: 2
print(findDuplicate([3, 1, 3, 4, 2]))  # Output: 3
print(findDuplicate([1, 1, 2]))        # Output: 1

File: DSA/IshikaGarg_500124973/Day_6_Q1.c
#include <stdio.h>

int findDuplicate(int nums[], int size) {
    int slow = nums[0];
    int fast = nums[0];
    do {
        slow = nums[slow];
        fast = nums[nums[fast]];
    } while (slow != fast);
    // Find the entrance to the cycle
    slow = nums[0];
    while (slow != fast) {
        slow = nums[slow];
        fast = nums[fast];
    }
    return slow;
}

int main() {
    int nums1[] = {1, 3, 4, 2, 2};
    printf("%d\n", findDuplicate(nums1, 5)); 
    return 0;
}
